frontend/src/app/projects/models/create-coworking-method/create-coworking-method.component.ts

```diff
@@ -156,52 +197,102 @@ export class CreateCoworkingMethodComponent implements OnInit {
     this.updateResultUrl();
     this.resetRevisions();
 
-    let urlsControls = this.form.controls.urls.controls;
-    urlsControls.inputUrl.updateValueAndValidity();
+    this.urls.inputUrl.updateValueAndValidity();
 
     if (changedInputUrl && hasAbsoluteUrlPrefix(changedInputUrl)) {
       let longestMatchingGitSetting =
         this.findLongestUrlMatchingGitSetting(changedInputUrl);
-
       if (longestMatchingGitSetting) {
-        urlsControls.baseUrl.setValue(longestMatchingGitSetting);
         this.selectedGitInstance = longestMatchingGitSetting;
+        this.urls.baseUrl.setValue(longestMatchingGitSetting);
       } else if (this.availableGitInstances.length) {
         this.selectedGitInstance = undefined;
-        this.form.controls.urls.controls.baseUrl.reset();
+        this.urls.baseUrl.reset();
       }
     }
   }
 
-  onSubmit(): void {
-    if (
-      this.form.valid &&
-      this.projectService.project &&
-      this.modelService.model
-    ) {
+  onCreateSubmit(): void {
+    if (this.form.valid) {
       this.sourceService
         .addGitSource(
-          this.projectService.project.name,
-          this.modelService.model.slug,
-          {
-            path: this.resultUrl,
-            username: this.form.value.credentials!.username,
-            password: this.form.value.credentials!.password,
-            revision: this.form.value.revision!,
-            entrypoint: this.form.value.entrypoint,
-          } as Source
+          this.projectService.project?.slug!,
+          this.modelService.model?.slug!,
+          this.createGitModelFromForm()
         )
         .subscribe(() => {
-          this.create.emit({ created: true });
+          if (this.asStepper) {
+            this.create.emit({ created: true });
+          } else {
+            this.location.back();
```

Comment:
The intended behaviour is not going back but redirecting to the previous page. This leads to unwanted consequences such as the possibility to go forward with the browser (when you would have liked to return to the page with the back arrow of your browser).

Use instead `this.router.navigate(['..'], { relativeTo: this.route });`.